src/lib/Analytics/Analytics.svelte

```diff
@@ -0,0 +1,77 @@
+<script context="module">
+  /**
+   * A function that logs a click event to Google Analytics.
+   * @param {HTMLElement} target The DOM element that triggered the event.
+   * @param {string} eventName A string in the format "event_name--event_specifics" where "event_name is the type of event, such as "map_click" or "dropdown_select" and "event_specifics" refers to the selected option, region, or specific function.
+   * @returns {void}
+   */
+  export function logClickToGA(target, eventName) {
+    const DEV = window.ui_dataviz_config.analytics_mode !== "production";
+    const eventData = {
+      dataviz_title: window.ui_dataviz_config.analytics_title,
+      dataviz_target: target,
+      dataviz_detail: eventName
+    }
+    if (!DEV) {
+      if (typeof window === "undefined" || !window.gtag) return;
+      window.gtag("event", "dataviz_click", eventData);
+    } else {
+      console.log("dataviz_click fired", eventData);
+    }
+  }
+</script>
+
+<script>
+  import { onMount } from "svelte";
+
+  /**
+   * A descriptive title for your project. Used to collect events within GA.
+   * @type {string}
+   */
+  export let title;
+
+  /**
+   * Set to false to skip initial pageview. Useful for projects that are primarily embedded on pages with their own analytics.
+   * @type {boolean} [true]
+   */
+  export let sendPageview = true;
+
+  /**
+   * If set to "development", logClickToGA will print debugging info rather than sending events to GA. Set to "production" to send actual events.
+   * @type {string} ["production"]
```

Comment:
`* @type {string} ["production" | "development"]`

Reply:
What do you think about defaulting to development and changing the docx examples to explicitly set the prop to production?

Reply:
I think we're on the same page about the right approach here but have slightly different ideas of how to acheive it. I think the component should do what it advertises by default without extra configuration. IE dev mode should be opt-in instead of opt-out. BUT i think for our projects, it makes sense to default to dev mode because we would assume that all projects begin that way. This is something we can implement on the project template side of things rather than enforcing it in the component. I'm picturing something like this in our project template:
```
<Analytics mode={import.meta.env.MODE} />
```
so that we can use the built-in way of specifying environment that we're already using. That way our projects would automatically send events when deployed to production and automatically skip them when running in dev mode and it wouldn't be up to the developer to remember to toggle this on and off while they're working.

This is also why I left the type definition more generic as a `{string}`, since at the very least we'd also have `staging` as an option.

Reply:
Yes love this approach!! All tracks with me!